internal/charmstore/store.go

```diff
@@ -1280,6 +1275,32 @@ func bundleCharms(data *charm.BundleData) ([]*charm.URL, error) {
 	return urls, nil
 }
 
+// MatchingInterfacesQuery returns a mongo query
+// that will find any charms that require any interfaces
+// in the required slice or provide any interfaces in the
+// provided slice.
+//
+// Development charms are never matched.
```

Comment:
It would be nice to know why they are not. I see value in matching them, but if we decided not to, let's add a commnet with the reason.